WcaOnRails/app/webpacker/components/DelegateTab/RoleForm.jsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+
+import { Form } from 'semantic-ui-react';
+
+import delegateStatus from '../../lib/helpers/delegate-status';
+import I18n from '../../lib/i18n';
+import { delegateUpdateDelegateUrl } from '../../lib/requests/routes.js.erb';
+import { fetchWithAuthenticityToken } from '../../lib/requests/fetchWithAuthenticityToken';
+
+const notDelegateOption = 'None';
+
+function RoleForm({
+  values,
+  regionList,
+  disabled,
+}) {
+  const { userId, ...delegateDetails } = values;
+  const [savedValues, setSavedValues] = React.useState(values.status ? delegateDetails : {});
+  const [formValues, setFormValues] = React.useState(values.status ? delegateDetails : {});
+
+  const updateDelegate = async () => {
+    const url = formValues.status ? delegateUpdateDelegateUrl(
+      userId,
+      formValues.status,
+      formValues.location,
+      formValues.regionId,
+    ) : delegateUpdateDelegateUrl(userId);
+    await fetchWithAuthenticityToken(url);
+    setSavedValues(formValues);
+  };
+
+  return (
+    <Form onSubmit={() => updateDelegate()}>
+      <Form.Dropdown
+        label={I18n.t('activerecord.attributes.user.delegate_status')}
+        fluid
+        selection
+        value={formValues.status || notDelegateOption}
+        disabled={disabled}
+        options={[{
+          key: notDelegateOption,
+          text: notDelegateOption,
+          value: notDelegateOption,
+        },
+        ...Object.keys(delegateStatus)
+          .map((status) => ({
+            key: status,
+            text: delegateStatus[status].name,
+            value: status,
+          }))]}
+        onChange={(_, { value }) => {
+          if (value === notDelegateOption) {
+            setFormValues({});
+          } else {
+            setFormValues({ ...formValues, status: value });
+          }
+        }}
+      />
+      {formValues.status && !delegateStatus[formValues.status].isLeadRole && (
+        <Form.Input
+          label={I18n.t('activerecord.attributes.user.location')}
+          value={formValues.location}
+          disabled={disabled}
+          onChange={(_, { value }) => setFormValues({ ...formValues, location: value })}
+        />
+      )}
+      {/* TODO: Add country field too after introduction of delegates table */}
+      {formValues.status && (
+        <Form.Dropdown
+          label={I18n.t('activerecord.attributes.user.region')}
+          placeholder="Select Region"
+          fluid
+          selection
+          value={formValues.regionId}
+          disabled={disabled || formValues.status === 'None'}
+          options={regionList.map((region) => ({
+            key: region.id,
+            text: region.name,
+            value: region.id,
+          }))}
+          onChange={(_, { value }) => setFormValues({ ...formValues, regionId: value })}
+        />
+      )}
+      <div style={{
```

Comment:
I think we usually keep the styling out of the jsx files? Not sure if that's important though.

Reply:
I didn't put it outside because it's just a single style and I didn't feel like creating a css file just for one style. If you think it's still important, please let me know, I'll move to a new file.